modules/apps/captcha/captcha-taglib/src/main/resources/META-INF/resources/captcha/simplecaptcha.jsp

```diff
@@ -31,33 +38,54 @@ String url = (String)request.getAttribute("liferay-captcha:captcha:url");
 
 	<aui:script>
 		var hasEventAttached = false;
+		var refreshCaptcha = document.getElementById('<portlet:namespace />refreshCaptcha');
+		var captcha = document.getElementById('<portlet:namespace />captcha');
+		var simpleCaptchaURL = '/o/captcha/v1.0/simple?portletId=<%= portletId %>';
 
 		function attachEvent() {
-			var refreshCaptcha = document.getElementById(
-				'<portlet:namespace />refreshCaptcha'
-			);
-
 			if (refreshCaptcha && !hasEventAttached) {
 				hasEventAttached = true;
-				refreshCaptcha.addEventListener('click', () => {
-					var url = Liferay.Util.addParams(
-						't=' + Date.now(),
-						'<%= HtmlUtil.escapeJS(url) %>'
-					);
-
-					var captcha = document.getElementById(
-						'<portlet:namespace />captcha'
-					);
-
-					if (captcha) {
-						captcha.setAttribute('src', url);
-					}
-				});
+				refreshCaptcha.addEventListener('click', handleCaptchaRefresh);
+			}
+		}
+
+		function handleCaptchaRefresh() {
+			if (<%= headlessAPI %>) {
+				getCaptchaSourceImage(simpleCaptchaURL);
+			} else {
+				var url = Liferay.Util.addParams('t=' + Date.now(), '<%= HtmlUtil.escapeJS(url) %>');
```

Comment:
for variable inside the function in js, it is better to use let

Reply:
Fixed it. Thanks!